src/dc_receive_imf.rs

```diff
@@ -877,10 +861,7 @@ async fn add_parts(
     // messages such as read receipts can be useful to detect
     // ephemeral timer support, but timer changes without visible
     // received messages may be confusing to the user.
-    if !*hidden
-        && !location_kml_is
-        && !is_mdn
-        && chat_id.get_ephemeral_timer(context).await? != ephemeral_timer
+    if !location_kml_is && !is_mdn && chat_id.get_ephemeral_timer(context).await? != ephemeral_timer
```

Comment:
if `chat_id` is set to `DC_CHAT_ID_TRASH` before, the last condition loads the ephemeral timer of the trash from the database and compare it to the given value; executing the condition below if there are differences. 

i do not think we should execute epehmeral timer instructions for the chat at all, so maybe add a check explicitly for `!chat_id.is_trash()` beforehand.

Reply:
I have added  a commit to address this issue. Also checking if there are any parts left instead of explicit `location.kml` check, because for `location.kml` messages there can still be a visible part. `is_mdn` check is not needed because MDNs have no visible parts and are assigned to the trash chat now.

Reply:
great, now all lgtm :)